src/main/java/de/tum/in/www1/artemis/service/scheduled/DataExportScheduleService.java

```diff
@@ -72,13 +72,18 @@ public void createDataExportsAndDeleteOldOnes() {
         var dataExportsToBeCreated = dataExportRepository.findAllToBeCreated();
         ExecutorService executor = Executors.newFixedThreadPool(10);
         dataExportsToBeCreated.forEach(dataExport -> executor.execute(() -> createDataExport(dataExport, successfulDataExports)));
+        executor.shutdown();
         var dataExportsToBeDeleted = dataExportRepository.findAllToBeDeleted();
         dataExportsToBeDeleted.forEach(this::deleteDataExport);
         Optional<User> admin = userService.findInternalAdminUser();
         if (admin.isEmpty()) {
             log.warn("No internal admin user found. Cannot send email to admin about successful creation of data exports.");
             return;
         }
+        if (!executor.awaitTermination(15, java.util.concurrent.TimeUnit.MINUTES)) {
+            log.warn("Data export creation did not finish within 15 minutes.");
+            executor.shutdownNow();
```

Comment:
Should this also trigger a failure mail?

Reply:
we already this discussed it, I just add it here: no it shouldn't because it isn't a failure, the creation is just postponed.